charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml

```diff
@@ -62,6 +62,20 @@ spec:
             - --logtostderr=true
             {{- end }}
             - --localhost=127.0.0.1
+            {{- if .Values.prometheusOperator.prometheusDefaultBaseImage }}
+            {{- if .Values.prometheusOperator.prometheusDefaultBaseImage.sha }}
+            - --prometheus-default-base-image={{ .Values.prometheusOperator.prometheusDefaultBaseImage.repository }}:{{ .Values.prometheusOperator.prometheusDefaultBaseImage.tag }}@sha256:{{ .Values.prometheusOperator.prometheusDefaultBaseImage.sha }}
+            {{- else }}
+            - --prometheus-default-base-image={{ .Values.prometheusOperator.prometheusDefaultBaseImage.repository }}:{{ .Values.prometheusOperator.prometheusDefaultBaseImage.tag }}
+            {{- end }}
+            {{- end }}
+            {{- if .Values.prometheusOperator.alertmanagerDefaultBaseImage }}
+            {{- if .Values.prometheusOperator.alertmanagerDefaultBaseImage.sha }}
+            - --alertmanager-default-base-image={{ .Values.prometheusOperator.alertmanagerDefaultBaseImage.repository }}:{{ .Values.prometheusOperator.alertmanagerDefaultBaseImage.tag }}@sha256:{{ .Values.prometheusOperator.alertmanagerDefaultBaseImage.sha }}
```

Comment:
Why do you force usage of the sha?
Could be optional.

Reply:
I am not forcing the sha here. I am applying it if  `.Values.prometheusOperator.alertmanagerDefaultBaseImage.sha` is set, otherwise, `--alertmanager-default-base-image={{ .Values.prometheusOperator.alertmanagerDefaultBaseImage.repository }}:{{ .Values.prometheusOperator.alertmanagerDefaultBaseImage.tag }}` will be applied

I took this part from the prometheusConfigReloaderImage part below actually